### PR TITLE
tsm1 file dump

### PR DESCRIPTION
**Pull request by jwilder**
This adds the ability to dump some lower level info about `tsm1` files.  It will dump index, block, fields, summary and compression data given a `tsm1` file.  It does this by reading the file directly and not through the engine.  Sample output can be seen [here]( https://gist.github.com/jwilder/785a03d6e64cb8119a7e).

Running this uncovered two compression issues:
* Monitoring data was stored with nanosecond precision but 10s apart so it was not compressed well.  This change monitoring data to record at seconds precision which allows us to use run-length encoding for timestamps.
* Timestamp encoder had a bug where run-length encoding could be used but we would use simple8b which only compresses better than run-length encoding in a few cases.

The `influx_inspect` code is not pretty, but the output is very useful.

### CODE REVIEW

**Comment by syepes** [comment]
This is going to be really useful, the only thing I see is that there should be some kind of option to limit/filter the output of the Index/Block section so it can me used in large files.

**Comment by pauldix** [comment]
+1

File: monitor/service.go [hunk 1]
```diff
@@ -368,7 +368,7 @@ func (m *Monitor) storeStatistics() {
 
 			points := make(models.Points, 0, len(stats))
 			for _, s := range stats {
-				points = append(points, models.NewPoint(s.Name, s.Tags, s.Values, time.Now()))
+				points = append(points, models.NewPoint(s.Name, s.Tags, s.Values, time.Now().Truncate(time.Second)))
```
Comment (otoolep) [inline monitor/service.go]: Why did you make this change @jwilder ?

Comment (jwilder) [inline monitor/service.go]: See PR description

Comment (otoolep) [inline monitor/service.go]: RTFPR. :-)

Thanks.

**Pull request merged**